kinto/core/storage/postgresql/__init__.py

```diff
@@ -533,9 +533,19 @@ def purge_deleted(self, collection_id, parent_id, before=None,
             placeholders['before'] = before
 
         with self.client.connect() as conn:
-            result = conn.execute(query.format_map(safeholders), placeholders)
+            result = conn.execute(delete_tombstones.format_map(safeholders), placeholders)
+            deleted = result.rowcount
+
+            # If purging everything from a parent_id, then clear timestamps.
+            if collection_id is None and before is None:
+                delete_timestamps = """
+                DELETE
+                FROM timestamps
+                WHERE {parent_id_filter}
+                """
+                conn.execute(delete_timestamps.format_map(safeholders), placeholders)
```

Comment:
Unless I am completely confused at this time of the day, the second query does not depend on the result of the first one, so we can safely execute those two queries separately. They will be executed in the same transaction anyway.